docs/api.md

```diff
@@ -2346,43 +2352,81 @@ Verify a session using a recovery code.
   <!--end-response-body-post-sessionverifyrecoverycode-remaining-->
 
 
-### Recovery keys
+### Recovery key
 
-#### POST /recoveryKeys
+#### POST /recoveryKey
 
 :lock: HAWK-authenticated with session token
-<!--begin-route-post-recoverykeys-->
+<!--begin-route-post-recoverykey-->
 Creates a new recovery key for a user.
 
 Recovery keys are one-time-use tokens
 that can be used to recover the user's kB
 if they forget their password.
 For more details, see the
 [recovery keys](recovery_keys.md) docs.
-<!--end-route-post-recoverykeys-->
+<!--end-route-post-recoverykey-->
 
 ##### Request body
 
 * `recoveryKeyId`: *validators.recoveryKeyId*
 
-  <!--begin-request-body-post-recoverykeys-recoveryKeyId-->
+  <!--begin-request-body-post-recoverykey-recoveryKeyId-->
   A unique identifier for this recovery key, derived from the key via HKDF.
-  <!--end-request-body-post-recoverykeys-recoveryKeyId-->
+  <!--end-request-body-post-recoverykey-recoveryKeyId-->
 
 * `recoveryData`: *validators.recoveryData*
 
-  <!--begin-request-body-post-recoverykeys-recoveryData-->
+  <!--begin-request-body-post-recoverykey-recoveryData-->
   An encrypted bundle containing the user's kB.
-  <!--end-request-body-post-recoverykeys-recoveryData-->
+  <!--end-request-body-post-recoverykey-recoveryData-->
 
 
-#### GET /recoveryKeys/{recoveryKeyId}
+#### GET /recoveryKey/:recoveryKeyId
 
 :lock: HAWK-authenticated with account reset token
-<!--begin-route-get-recoverykeysrecoverykeyid-->
+<!--begin-route-get-recoverykeyrecoverykeyid-->
 Retrieve the account recovery data associated with the given recovery key.
-<!--end-route-get-recoverykeysrecoverykeyid-->
+<!--end-route-get-recoverykeyrecoverykeyid-->
 
+##### Response body
+
+* `recoveryData`: *string*
+
+  <!--begin-response-body-post-recoverykeyrecoverykeyid-recoverydata-->
+  
+  <!--end-response-body-post-recoverykeyrecoverykeyid-recoverydata-->
+
+#### POST /recoveryKey/exists
+:lock: HAWK-authenticated with session token
+<!--begin-route-post-recoverykeyexists-->
+This route checks to see if given user has setup an account recovery key.
+When used during the password reset flow, an email can be provided (instead
+of a sessionToken) to check for the status. However, when
+using an email, the request is rate limited.
```

Comment:
During the reset flow, is the call completely unauthenticated, or could we authenticate it with e.g. the accountResetToken?

Reply:
Yea, currently in the reset flow the call is unauthenticated. This was done to ensure that the `accountResetToken` didn't get consumed prematurely. Which my understanding was that it could only been used once?

After speaking with Ana, we suspected that it might take a while for a user to get their recovery key, especially if they have to track down a friend. A user might click the email link, get redirected to account recovery, then close window and re-click email link once they have information.

I was trying to model this endpoint after [account status](https://github.com/mozilla/fxa-auth-server/blob/6d7942de0a10ff8102fdb165fb9503dd71ea1258/lib/routes/account.js#L765) but make it a little more efficient and use only one route. Is another way we might be able to do this check but authenticated?

Reply:
>  A user might click the email link, get redirected to account recovery, then close window
> and re-click email link once they have information.

Ah, interesting!  We definitely need to be careful here then; the link from the email contains the "password forgot token" and a verification code, which is indeed single-use.  

  https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#post-passwordforgotverify_code

What are the steps that the content-server will take here?  From your description I'm inferring something like:

1. Load the link from the reset email, but don't consume the `passwordForgotToken` yet.
2. Check whether recovery key exists for the target account.  If not, go to X.
3. Prompt the user to enter their recovery key.  The user may bounce out at this point and re-enter at (1).  We can't yet check whether they're entered a valid key.
4. Consume the `passwordForgotToken` to obtain the `accountResetToken`.
5. Use the `acccountResetToken` to redeem the recovery key and get the recovery data.  This also checks the recovery key for validity; will they bounce back to (1) here if it was invalid and they need to try again?
6. Prompt for the new password.
7. Use the `accountResetToken` to set the new password and `wrapKb`, consuming the `accountResetToken`.

> Is there another way we might be able to do this check but authenticated?

I think it's fine as-is if it helps with the flow, just wanted to check.